controllers/medusa/medusarestorejob_controller.go

```diff
@@ -262,26 +264,33 @@ func (r *MedusaRestoreJobReconciler) podTemplateSpecUpdateComplete(ctx context.C
 }
 
 // prepareRestore prepares the MedusaRestoreMapping for the restore operation.
-// It uses the Medusa client to get the host map for the restore operation, using the Medusa standalone service to get the backup topology.
-func (r *MedusaRestoreJobReconciler) prepareRestore(ctx context.Context, request *medusa.RestoreRequest) (*medusav1alpha1.MedusaRestoreMapping, error) {
-	medusaClient, err := r.ClientFactory.NewClient(ctx, medusaServiceUrl(
-		cassdcapi.CleanupForKubernetes(request.Datacenter.Spec.ClusterName),
-		request.Datacenter.SanitizedName(),
-		request.Datacenter.Namespace))
-	if err != nil {
-		return nil, err
-	}
-	restoreHostMap, err := medusa.GetHostMap(request.Datacenter, *request.RestoreJob, medusaClient, ctx)
+// It uses the Medusa client to get the host map for the restore operation, using the first pod answering on the backup sidecar port.
+func (r *MedusaRestoreJobReconciler) prepareRestore(ctx context.Context, request *medusa.RestoreRequest, logger logr.Logger) (*medusav1alpha1.MedusaRestoreMapping, error) {
+	pods, err := medusa.GetCassandraDatacenterPods(ctx, request.Datacenter, r, request.Log)
 	if err != nil {
+		logger.Error(err, "Failed to get datacenter pods")
 		return nil, err
 	}
 
-	medusaRestoreMapping, err := restoreHostMap.ToMedusaRestoreMapping()
-	if err != nil {
-		return nil, err
-	}
+	for _, pod := range pods {
+		addr := net.JoinHostPort(pod.Status.PodIP, fmt.Sprint(shared.BackupSidecarPort))
+		if medusaClient, err := r.ClientFactory.NewClient(ctx, addr); err != nil {
+			logger.Error(err, "Failed to create Medusa client", "address", addr)
+		} else {
+			restoreHostMap, err := medusa.GetHostMap(request.Datacenter, *request.RestoreJob, medusaClient, ctx)
```

Comment:
Probably another issue here. If GetHostMap fails due to network issues, you surely want to continue onto the next pod again don't you? Right now this returns...